File: ems-server/src/main/kotlin/io/cloudflight/ems/exception/I18nValidationError.kt
package io.cloudflight.ems.exception

import org.springframework.http.HttpStatus

data class I18nValidationError(
    val i18nKey: String? = null,
    val i8nArguments: List<String>? = null,
    val httpStatus: HttpStatus,
    val i18nFieldErrors: Map<String, I18nFieldError>? = null,
    val additionalInfo: String? = null
) : Exception()

File: ems-ui/src/app/common/validation/i18n-validation-error.ts
import {I18nFieldError} from './i18n-field-error';

export class I18nValidationError {
  i18nKey?: string;
  i8nArguments?: string[];
  httpStatus: number;
  i18nFieldErrors?: { [key: string]: I18nFieldError };
  additionalInfo?: string;
}
